pkg/agent/storage/backcompat.go

```diff
@@ -0,0 +1,60 @@
+package storage
+
+import (
+	"crypto/x509"
+	"path/filepath"
+)
+
+func Backcompat(dir string) (Storage, error) {
+	j, err := JSONFile(filepath.Join(dir, "data.json"))
+	if err != nil {
+		return nil, err
+	}
+	return &backcompat{
+		jsonFile: j,
+		legacy:   LegacyDir(dir),
+	}, nil
+}
+
+type backcompat struct {
+	jsonFile Storage
+	legacy   Storage
+}
+
+func (b *backcompat) LoadBundle() ([]*x509.Certificate, error) {
+	return b.jsonFile.LoadBundle()
+}
+
+func (b *backcompat) StoreBundle(bundle []*x509.Certificate) error {
+	if err := b.jsonFile.StoreBundle(bundle); err != nil {
+		return err
+	}
+	if err := b.legacy.StoreBundle(bundle); err != nil {
+		return err
```

Comment:
If there is an error here, the situation would be well handled if we check for modification time when we are populating, right?

Reply:
This is tricky. I need to think through this some more.